lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp

```diff
@@ -530,8 +527,9 @@ class BasicBlockCloner : public SILClonerWithScopes<BasicBlockCloner> {
     for (auto *arg : bb->getArguments()) {
       // Create a new argument and copy it into the ValueMap so future
       // references use it.
-      ValueMap[arg] = newBB->createPhiArgument(
-          arg->getType(), arg->getOwnershipKind(), arg->getDecl());
+      updateValueMap(arg, newBB->createPhiArgument(arg->getType(),
```

Comment:
Why not just remove `updateValueMap` and use `mapValue` directly?

Reply:
mapValue is protected, but we need to invoke that from outside the class [unrollLoopBodyOnce](https://github.com/apple/swift/blob/359de5b28af78236a97311db8aa3a8d4f1442400/lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp#L1525).